pkg/api/validation/validation.go

```diff
@@ -1210,15 +1219,42 @@ func ValidateReplicationControllerSpec(spec *api.ReplicationControllerSpec) errs
 	return allErrs
 }
 
+// ValidateDaemonController tests if required fields in the daemon controller are set.
+func ValidateDaemonController(controller *api.DaemonController) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, ValidateObjectMeta(&controller.ObjectMeta, true, ValidateReplicationControllerName).Prefix("metadata")...)
+	allErrs = append(allErrs, ValidateDaemonControllerSpec(&controller.Spec).Prefix("spec")...)
+	return allErrs
+}
+
+// ValidateDaemonControllerUpdate tests if required fields in the daemon controller are set.
+func ValidateDaemonControllerUpdate(oldController, controller *api.DaemonController) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	allErrs = append(allErrs, ValidateObjectMetaUpdate(&oldController.ObjectMeta, &controller.ObjectMeta).Prefix("metadata")...)
+	allErrs = append(allErrs, ValidateDaemonControllerSpec(&controller.Spec).Prefix("spec")...)
+	return allErrs
+}
+
+// ValidateDaemonControllerSpec tests if required fields in the daemon controller spec are set.
+func ValidateDaemonControllerSpec(spec *api.DaemonControllerSpec) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if spec.Template == nil {
+		allErrs = append(allErrs, errs.NewFieldRequired("template"))
+	} else {
+		allErrs = append(allErrs, ValidatePodTemplateSpec(spec.Template).Prefix("template")...)
+		if spec.Template.Spec.RestartPolicy != api.RestartPolicyAlways {
+			allErrs = append(allErrs, errs.NewFieldNotSupported("template.restartPolicy", spec.Template.Spec.RestartPolicy))
```

Comment:
If the user doesn't specify a restart policy, then the pod spec will assign the default restart policy: RestartPolicyAlways. The daemon controller will then pass the validation test. The only issue is if the user explicitly sets the restart policy to something other than always. Then the validation will fail.

What aspect of the functionality should I change?